descwl_coadd/coadd.py

```diff
@@ -1331,202 +793,85 @@ def _finish_init(self):
-            world_origin=galsim.PositionD(0, 0),
-        ),
-        world_origin=world_origin,
-        units=galsim.arcsec,
-    )
-
-    return make_stack_wcs(tan_wcs)
-
-
-def repair_exp(exp, show=False, border_size=None):
-    """
-    run a RepairTask, currently not sending defects just
-    finding cosmics and interpolating them
+    coadd_wcs: DM wcs
+        The coadd wcs
+    coadd_bbox: geom.Box2I
+        The bounding box for the coadd within larger wcs system
+    psf_dims: tuple
+        The dimensions of the psf
 
-    Parameters
-    ----------
-    exp:
-        an Exposure from the stack
-    border_size: bool
-        border size to zero
-    show: bool
-        If True, show the image
+    Returns
+    -------
+    A DM SkyWcs
     """
-    from lsst.pipe.tasks.repair import RepairTask, RepairConfig
-
-    if show:
-        print('before repair')
-        vis.show_image(exp.image.array)
-
-    if border_size is not None:
-        b = border_size
-        ny, nx = exp.image.array.shape
-        exp.image.array[0:b, :] = 0
-        exp.image.array[ny-b:, :] = 0
-        exp.image.array[:, 0:b] = 0
-        exp.image.array[:, nx-b:] = 0
+    cy, cx = (np.array(psf_dims)-1)/2
+    psf_crpix = geom.Point2D(x=cx, y=cy)
 
-    repair_config = RepairConfig()
-    repair_task = RepairTask(config=repair_config)
-    repair_task.run(exposure=exp)
-
-    if show:
-        print('after repair')
-        vis.show_image(exp.image.array)
-
-
-def add_cosmics_to_noise(*, exp, noise_exp, value=1.0e18):
-    """
-    add fake cosmics to the noise exposure wherever
-    they are set in the real image
+    coadd_pixcen, coadd_skycen = get_coadd_center(
+        coadd_wcs=coadd_wcs,
+        coadd_bbox=coadd_bbox,
+    )
+    coadd_cd_matrix = coadd_wcs.getCdMatrix(coadd_pixcen)
 
-    TODO: get a realistic value for the real data
-    """
+    # coadd_skycen = coadd_wcs.getSkyOrigin()
+    # coadd_cd_matrix = coadd_wcs.getCdMatrix(coadd_wcs.getPixelOrigin())
 
-    CR = 2**afw_image.Mask.getMaskPlane('CR')  # noqa
-    w = np.where((exp.mask.array & CR) != 0)
-    print('pixels for cosmics:', w[0].size)
-    if w[0].size > 0:
-        noise_exp.image.array[w] = value
+    return makeSkyWcs(
+        crpix=psf_crpix,
+        crval=coadd_skycen,
+        cdMatrix=coadd_cd_matrix,
+    )
 
 
-def add_badcols_to_noise(*, exp, noise_exp):
+def make_logger(name, loglevel):
     """
-    Set bad cols in the noise image based on the real
-    image bits
+    make a logger with the specified loglevel
     """
-
-    BAD = 2**afw_image.Mask.getMaskPlane('BAD')  # noqa
-    w = np.where((exp.mask.array & BAD) != 0)
-    print('pixels for badcols:', w[0].size)
-    if w[0].size > 0:
-        noise_exp.image.array[w] = exp.image.array[w]
+    logger = lsst.log.getLogger(name)
+    logger.setLevel(getattr(lsst.log, loglevel.upper()))
+    return logger
 
 
-def zero_bits(*, image, noise, mask, flags):
+def zero_bits(image, noise, mask, flags):
```

Comment:
Doc string here or add an _ to the function name?

Reply:
will add doc